### PR TITLE
Use Gazebo 8 APIs to avoid deprecation warnings.

### PR DESCRIPTION
**Pull request by esteve**
This PR uses the non-deprecated APIs in Gazebo 8.

### CODE REVIEW

**Comment by esteve** [comment]
I'm not sure if you like this level of granularity (I find all these `#if (GAZEBO_MAJOR_VERSION >= 8)` a bit ugly) or you'd prefer a less invasive refactor.

File: hector_gazebo_plugins/package.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
   <!-- Dependencies needed to compile this package. -->
   <build_depend>roscpp</build_depend>
-  <build_depend>libgazebo7-dev</build_depend>
+  <build_depend>gazebo_ros</build_depend>
```
Comment (meyerj) [inline hector_gazebo_plugins/package.xml]: Shouldn't this be the replaced by the new `gazebo_dev` package introduced in https://github.com/ros-simulation/gazebo_ros_pkgs/pull/571? Same as in https://github.com/ros-simulation/gazebo_ros_pkgs/blob/kinetic-devel/gazebo_plugins/package.xml.

**Comment by meyerj** [comment]
Thanks for the patches. I think the `#if` statements are fine and still much better than to fork and maintain separate branches per Gazebo release or to introduce adapters.

**Comment by esteve** [comment]
@meyerj I just changed the plugin to depend on `gazebo_dev` https://github.com/tu-darmstadt-ros-pkg/hector_gazebo/pull/44/commits/5387785c4ab85e16f0cf1ad1ce45cc765df620f3

I tested the plugins with Gazebo 9 a few days ago, and it turns out that the deprecated APIs ended up being removed in the current release.

**Comment by naoki-mizuno** [comment]
I've also successfully compiled this PR on my Gazebo9!

**Pull request merged**